### PR TITLE
Added privileges so that only instructors can endorse a post

### LINKED ISSUE #20: Ensuring that the ‘endorse’ option appears only for instructors that are logged in and not for students
Estimated Effort - 
High: We are not sure how difficult it is in NodeBB to manage privileges.

Acceptance Criteria-
When I log in as an instructor, I should see the Endorse option on the Question & Answer category. However, when I log in as a student, I should not be able to see the Endorse option. 

Component of #8

### PR DESCRIPTION
**Pull request by Malika1109**
Added function isInstructor to the privileges of a user so that I can check if a user is an instructor or not. 

Resolves #20

### CODE REVIEW

**Comment by GulnazSerikbay** [comment]
LGTM

File: src/posts/actions.js
Comment (GulnazSerikbay) [inline src/posts/actions.js:1]: console.log can be removed

File: src/posts/actions.js [hunk 1]
```diff
@@ -3,23 +3,59 @@
...
+            console.log("set to false");
+
+            return {
+                postId: { pid }, // Adjust the structure to match your needs
+                isEndorsed: false, // Set this value accordingly
+            };
+        }
+        throw new Error("User is not an instructor. Cannot endorse.");
+    };
+
```
Comment (GulnazSerikbay) [inline src/posts/actions.js:51]: understandable code and needed methods!

File: test/api.js [hunk 2]
```diff
@@ -599,7 +601,8 @@ describe("API", async () => {
                 it(`${_method.toUpperCase()} ${path}: response status code should match one of the schema defined responses`, () => {
                     // HACK: allow HTTP 418 I am a teapot, for now   👇
```
Comment (GulnazSerikbay) [inline test/api.js:602]: what does this mean, unclear here, but looks fine

File: test/posts.js [hunk 3]
```diff
@@ -241,101 +362,182 @@ describe('Post\'s', () => {
...
-                assert.ifError(err);
-                assert(data.posts.display_edit_tools);
-                assert(data.posts.display_delete_tools);
-                assert(data.posts.display_moderator_tools);
-                assert(data.posts.display_move_tools);
-                done();
-            });
+        it("should load post tools", (done) => {
+            socketPosts.loadPostTools(
+                { uid: globalModUid },
```
Comment (GulnazSerikbay) [inline test/posts.js:513]: nice tests, thanks!

**Review by GulnazSerikbay — APPROVED** [review]
Looks good, overall! 
Can clean up the code a bit in the future, but merging :)

File: test/api.js (hunk 2 above)
Comment (Malika1109) [reply test/api.js:602]: Hey, this was already in the legacy code, not added by me

**Pull request merged**